system/default_ad_api_helpers/automatic_pose_initializer/src/automatic_pose_initializer.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "automatic_pose_initializer.hpp"
+
+#include <memory>
+
+namespace automatic_pose_initializer
+{
+
+AutomaticPoseInitializer::AutomaticPoseInitializer() : Node("automatic_pose_initializer")
+{
+  const auto adaptor = component_interface_utils::NodeAdaptor(this);
+  group_cli_ = create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
+  adaptor.init_cli(cli_initialize_, group_cli_);
+  adaptor.init_sub(sub_state_, [this](const State::Message::ConstSharedPtr msg) { state_ = *msg; });
+
+  const auto period = rclcpp::Rate(1.0).period();
```

Comment:
As far as I tried on my PC, by setting the rate to higher value (e.g. `rclcpp::Rate(2.0)`), the `automatic_pose_initializer` started keep requesting for the pose initialization infinitely. It seems that the `timer_` was kept called, and this blocked `sub_state_` to update the `state_.state` from `UNINITIALIZED` to `INITIALIZED`.

Would you check if this happens in your PC, too? And if so, do you have any idea how to fix this issue?

Reply:
@isamu-takagi  https://github.com/autowarefoundation/autoware.universe/issues/1750